js/utils.js

```diff
@@ -45,6 +45,71 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
     return parseFloat(value);
   }
 
+  function getFilesInfo(options) {
+    var entry = options.entryData;
+    var detailViewFileOptions = _.filter(options.detailViewOptions, function(option) {
+      return option.editable && option.type === 'file';
+    });
+
+    var formFilesInfoInDetailViewOptions = detailViewFileOptions.map(function(detailViewFileOption) {
+      return new Promise(function(resolve) {
+        var label = '';
+        var labelEnabled = true;
+        var files = typeof entry.originalData[detailViewFileOption.column] === 'string' ?
+          splitByCommas(entry.originalData[detailViewFileOption.column]) :
+          entry.originalData[detailViewFileOption.column];
+
+        if (!files) {
+          return resolve();
+        }
+
+        switch (detailViewFileOption.fieldLabel) {
+          case 'column-name':
+            if (detailViewFileOption.column !== 'custom') {
+              label = detailViewFileOption.column;
+            }
+
+            break;
+          case 'custom-label':
+            label = new Handlebars.SafeString(Handlebars.compile(detailViewFileOption.customFieldLabel)(entry.originalData));
+
+            break;
+          case 'no-label':
+            labelEnabled = false;
+
+            break;
+          default:
+            break;
+        }
+
+        var filesDetailViewInfo = Promise.all(files.map(function(fileUrl) {
+          var fileId = Fliplet.Media.isRemoteUrl(fileUrl)[2];
+
+          return Fliplet.Media.Files.get(fileId).then(function(file) {
+            return {
+              name: file.name,
+              size: file.size,
```

Comment:
@YaroslavOvdii We need to show the file size, not, in the case of images, the image size.

![image](https://user-images.githubusercontent.com/290733/106898279-e4fbff80-66eb-11eb-9cb4-0b683e811317.png)

The endpoint response currently seems to be returning the image size, not file size. Can you confirm what this is displaying at the moment?

Reply:
@tonytlwu Actually we are receiving a different answer from the API.
In our case, size is always `null`.
We are using `Fliplet.Media.Files.get(fileId)` to get file info.
![image](https://user-images.githubusercontent.com/53430352/106899402-fc43e880-66fd-11eb-8792-f40ad8041002.png)

Reply:
@YaroslavOvdii I believe that refers to image size, which means the `null` response implies that image size isn't always available. We also need file size information, which appears to be missing.

@squallstar How much time do we need to add file size to the endpoint response? Note that [this thread](https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/517#discussion_r564619192) suggests that need the information in bulk.

Reply:
@tonytlwu we already return that whenever available under `metadata.size` (in bytes) but it's only available for files uploaded via the media endpoints (e.g. from the file manager).

Reply:
@squallstar Excellent. Then it's just a matter of whether it exists or not.

@YaroslavOvdii See @squallstar's comment above that you need to use `metadata.size` for the file size.